krabs/krabs/filtering/comparers.hpp

```diff
@@ -80,10 +86,20 @@ namespace krabs { namespace predicates {
             template <typename Iter1, typename Iter2>
             bool operator()(Iter1 begin1, Iter1 end1, Iter2 begin2, Iter2 end2) const
             {
-                auto r1 = boost::make_iterator_range(begin1, end1);
-                auto r2 = boost::make_iterator_range(begin2, end2);
+                const auto dist1 = std::distance(begin1, end1);
```

Comment:
As above, this works for me here:
```C++
const auto dist1 = std::distance(begin1, end1);
const auto dist2 = std::distance(begin2, end2);
return dist1 >= dist2 && std::equal(end1 - dist2, end1, begin2, Comparer());
```

Reply:
As above, avoid the iterator math to avoid dependency on RandomAccess iterators.

Reply:
I guess I was thinking of these functors being used only on strings and not arbitrary, non-random access containers (in fact I'm not sure why I even used std::distance() instead of just endN-beginN). I take the point that avoiding the assumption of random access makes these as general as possible, and that a good template should be as general as possible. But when it comes to string comparison in particular we don't want a quest for generality to be (a) slower and (b) harder to understand/maintain.

Reply:
Those functions are idiomatic c++ and are optimized away correctly for the concrete instantiation of the template. They are not slower.